src/library_syscall.js

```diff
@@ -222,6 +223,16 @@ var SyscallsLibrary = {
       else assert(high === -1);
 #if SYSCALL_DEBUG
       err('    (i64: "' + low + '")');
+#endif
+      return low;
+    },
+    // Like get64, but read high first and low second.
+    get64hilo: function() {
+      var high = SYSCALLS.get(), low = SYSCALLS.get();
+      if (low >= 0) assert(high === 0);
```

Comment:
could this check with a `>> 31` like in library.js? or is there a reason to do it differently here?

Reply:
This was just copied from the existing `get64`. I've now updated both versions to use the `>> 31` check.